src/pybind/mgr/dashboard/HACKING.rst

```diff
@@ -332,6 +332,87 @@ vs. ``it('image edit test' () => ...)``. As shown, the first example makes gramm
 prefix whereas the second message does not.``it()`` should describe what the individual test is doing and
 what it expects to happen.
 
+Differences between Frontend Unit Tests and End-to-End (E2E) Tests / FAQ
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+General introduction about testing and E2E/unit tests 
+
+
+What are E2E/unit tests designed for?
+.....................................
+
+E2E test:
+
+"Protractor is an end-to-end test framework for Angular and AngularJS applications.
+Protractor runs tests against your application running in a real browser,
+interacting with it as a user would." `(src) <http://www.protractortest.org/#/>`__
+
+It requires a fully functional system and tests the interation of all components
+of the application (Ceph, back-end, front-end).
+E2E tests are designed to mimic the behavior of the user when interacting with the application
+- for example when it comes to workflows like creating/editing/deleting an item.
+Also the tests should verify that certain items are displayed as a user would see them
+when clicking through the UI.
+
+Angular Unit Tests:
+
+Unit tests, as the name suggests, are tests for smaller units of the code.
+Those tests are designed for testing all kinds of Angulars' components (e.g. services, pipes etc.).
+They do not require a connection to the backend and hence are independ of it,
+although the expected result of the backend can be verified using only the frontend.
+Data is either mocked or, in a simple case, contains a static input, a function call
+and an expected static output. More complex examples include the state of
+a component (attributes of the component class),that define how the output changes
+according to the given input.
+
+Which E2E/unit tests are considered to be valid?
+................................................
+
+This is not easy to answer, but new tests that are written in the same way as already existing
+dashboard tests should generally be considered valid.
+Unit tests should focus on the component to be tested.
+This is either an Angular component, directive, service, pipe, etc.
+
+E2E tests should focus on testing the functionality of the whole application.
+Approximately half of the E2E tests should verify the correctness of user visible elements.
```

Comment:
We've seen your comment, but for @p-se and me it still makes sense to have tests that verify the correctness of elements (e.g. in a form / form validation) - because in the end, it's still something the user can see and will take a look at.
Let's say you want to write a test which verifies the creation of a pool - I would not want to see "all" form (validation) tests (which actually do not create anything) in the same step as the test that actually verifies the creation of a pool. I think that would be too much distraction and kind of "off-topic" there.

Reply:
Please make sure to point out that it's about the current practice.

```suggestion
Currently approximately half of the E2E tests verify the correctness of user visible elements.
```

Reply:
Updated it to match the Testing Pyramid.